src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -70,6 +84,76 @@ public static void initIndexes() throws Exception {
         Collections.sort(indexes);
     }
     
+    @AfterClass
+    public static void tearDownStatics() {
+        indexes = null;
+        indicesDir = null;
+    }
+    
+    @Override
+    public Settings nodeSettings(int ord) {
+        return ImmutableSettings.builder()
+            .put(Node.HTTP_ENABLED, true) // for _upgrade
+            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
+            .build();
+    }
+    
+    void setupDataLoadingNodes() throws Exception {
+        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
+        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
```

Comment:
Well, obviously this is not about the line of codes.  

It better integrates with the internal cluster + the NodeEnvironment choice's of data paths. No hard coded reference to it or the node ordinal. It creates node data where it is supposed to be, as opposed to a random place somewhere etc (I can come up with more if need be :) ) . 

If in the future we want to introduce any randomness (which imho is a good thing here because you want to see that bwc works under different node environments, but that's another discussion), we practically get it for free.